src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/IdentifierFilter.php

```diff
@@ -0,0 +1,164 @@
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+
+            case Operators::DOES_NOT_CONTAIN:
+                $mustNotClause = [
+                    'query_string' => [
+                        'default_field' => self::IDENTIFIER_KEY,
+                        'query'         => '*' . $value . '*',
+                    ],
+                ];
+
+                $filterClause = [
+                    'exists' => ['field' => self::IDENTIFIER_KEY],
+                ];
+
+                $this->searchQueryBuilder->addMustNot($mustNotClause);
+                $this->searchQueryBuilder->addFilter($filterClause);
+                break;
+
+            case Operators::EQUALS:
+                $clause = [
+                    'term' => [
+                        self::IDENTIFIER_KEY => $value,
+                    ],
+                ];
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+
+            case Operators::NOT_EQUAL:
+                $mustNotClause = [
+                    'term' => [
+                        self::IDENTIFIER_KEY => $value,
+                    ],
+                ];
+
+                $filterClause = [
+                    'exists' => [
+                        'field' => self::IDENTIFIER_KEY,
+                    ],
+                ];
+                $this->searchQueryBuilder->addMustNot($mustNotClause);
+                $this->searchQueryBuilder->addFilter($filterClause);
+                break;
+
+            case Operators::IN_LIST:
+                $clause = [
+                    'terms' => [
+                        self::IDENTIFIER_KEY => $value
+                    ]
+                ];
+
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+
+            case Operators::NOT_IN_LIST:
+                $clause = [
+                    'terms' => [
+                        self::IDENTIFIER_KEY => $value
+                    ]
+                ];
+
+                $this->searchQueryBuilder->addMustNot($clause);
+                break;
+
+            default:
+                throw InvalidOperatorException::notSupported($operator, static::class);
+        }
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsField($field)
+    {
+        return in_array($field, $this->supportedFields) || $field === $this->attributeRepository->getIdentifierCode();
+    }
+
+    /**
+     * Checks the identifier is a string or an array depending on the operator
+     *
+     * @param string $field
+     * @param string $operator
+     * @param mixed  $value
+     *
+     */
+    protected function checkValue($field, $operator, $value)
+    {
+        if (Operators::IN_LIST === $operator || Operators::NOT_IN_LIST === $operator) {
+            FieldFilterHelper::checkArray($field, $value, self::class);
+        } elseif (in_array($operator, $this->supportedOperators)) {
```

Comment:
But it could be just an else condition instead of an elesif as you check the existence of the operator on the switch of the addFieldFilter ;-)